### ISSUE TITLE
[jit] ScriptModule with Dropout layers can't be set to eval

### ISSUE THREAD

**Issue by Evpok** [pytorch/pytorch#13991]
## 🐛 Bug

Adding a `Dropout` layer to a `jit.ScriptModule` makes it impossible to set that module to eval mode.

## MWE

```python
import torch
import torch.jit

class Net(torch.jit.ScriptModule):
	def __init__(self):
		super().__init__()
		self.linear = torch.nn.Linear(10, 10)
		self.dropout = torch.nn.Dropout(0.1)

	def forward(self, x):
		x = self.dropout(x)
		x = self.linear(x)
		return x

n = Net()
n.eval()
```
Fails with

```
RuntimeError: attempting to re-assign constant 'training'
```

## Environment
PyTorch version: 1.0.0.dev20181109
Is debug build: No
CUDA used to build PyTorch: None

OS: KDE neon Developer Edition
GCC version: (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
CMake version: version 3.10.2

Python version: 3.7
Is CUDA available: No

Versions of relevant libraries:
[pip3] numpy (1.15.4)
[pip3] pytorch-ignite (0.1.1)
[pip3] torch-nightly (1.0.0.dev20181109)
[pip3] torchtext (0.3.1)

**Comment by wanchaol** [comment]
@Evpok Yes `model.eval()` for ScriptModule is not supported yet. You can still use the dropout layer for training. For inference, a workaround would be: stay in the training mode and set the batch_size = 1, that will make it behave the same as `model.eval()`

**Comment by soumith** [comment]
@wanchaol dropout with batch_size = 1 wont make it behave the same as `model.eval()`. Maybe you are thinking about BatchNorm (which is also not the case that batch_size = 1 will be the same as model.eval())

**Comment by wanchaol** [comment]
> @wanchaol dropout with batch_size = 1 wont make it behave the same as `model.eval()`. Maybe you are thinking about BatchNorm (which is also not the case that batch_size = 1 will be the same as model.eval())

sorry I was wrong, I actually mean set `dropout_prob=0` when inference

**Comment by soumith** [comment]
similar to #14027

**Comment by draplater** [comment]
I think it's related to the design of ScriptModule. the "training" flag is seen as a constant, so it cannot be changed after the module is built. I'm wondering how pytorch team will solve this problem.

**Comment by driazati** [comment]
Fixed in #14719
